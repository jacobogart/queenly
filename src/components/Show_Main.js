import React, { Component } from "react";

// * CSS imports
import "../css/Main_Info.css";

export default class Show_Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: this.getLocation()
    }
  }

  stringifyHosts = host => {
    if (host) {
      switch (host.length) {
        case 0:
          return null;
        case 1:
          return `Host: ${host}`;
        case 2:
          return `Hosts: ${host.join(" & ")}`;
        default:
          return `Hosts: ${host.join(" , ")}`;
      }
    } else {
      return null;
    }
  };

  stringifyReoccuring = reoccuring => {
    return reoccuring ? "Every" : "Only this";
  };

  convertWhen = (dayOfWeek, frequency, reoccuring) => {
    let howOften = this.stringifyReoccuring(reoccuring);

    return `${howOften} ${dayOfWeek.dayOfWeek} on the ${
      frequency.frequency
    } of the month.`;
  };

  convertTime = startTime => {
    // let standardTime = startTime.startTime.toString().split('');
    // if (startTime.startTime >= 1300) {
    //   console.log('stand', standardTime)
    //   let hours = parseInt(standardTime.splice(0, 2).join(''));
    //   standardTime = standardTime.splice(0, 2, (hours - 12).split(''))
    //   console.log('hours', hours)
    //   console.log(standardTime)
    // }
  };

  // convertTime = (startTime) => {
  //   console.log('time: ', startTime.startTime.toString().split('').splice(0,0, ':'));
  //   let time = startTime.startTime.split('').splice(1,0, ':'); // your input
  //   console.log(time);
  //   time = time.split(':'); // convert to array

  //   // fetch
  //   var hours = Number(time[0]);
  //   var minutes = Number(time[1]);

  //   // calculate
  //   var timeValue;

  //   if (hours > 0 && hours <= 12) {
  //     timeValue = "" + hours;
  //   } else if (hours > 12) {
  //     timeValue = "" + (hours - 12);
  //   } else if (hours == 0) {
  //     timeValue = "12";
  //   }

  //   timeValue += (minutes < 10) ? ":0" + minutes : ":" + minutes;  // get minutes
  //   timeValue += (hours >= 12) ? " P.M." : " A.M.";  // get AM/PM
  // }

  getLocation = () => {
    let location = this.props.bars.find(bar => 
      bar.shows.find(show => 
        show.name.includes(this.props.cardData.name)
      )
    );

    return location.name;
  }

  goToBar = () => {
    this.props.selectSearchResult(this.state.location);
  }

  render() {
    let {
      name,
      notes,
      host,
      dayOfWeek,
      frequency,
      startTime,
      reoccuring
    } = this.props.cardData;

    return (
      <main className="Bar-Main Main_Info">
        <article className="Content">
          <h2>{name}</h2>
          {!notes ? null : (
            <div>
              <h4>About:</h4>
              <p className="notes">{notes}</p>
            </div>
          )}
          <p className="hosts">{this.stringifyHosts(host)}</p>
          <p>When: </p>
          <p className="showtime">
            {this.convertWhen({ dayOfWeek }, { frequency }, { reoccuring })}{" "}
          </p>
          <p className="time">At: {this.convertTime({ startTime })}</p>
          <p className="location" onClick={this.goToBar}> Where: {this.state.location}</p>
        </article>
      </main>
    );
  }
}
