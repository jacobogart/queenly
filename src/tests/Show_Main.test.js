import React from 'react';
import ReactDOM from 'react-dom';
import Show_Main from '../components/Show_Main.js';
import { shallow } from 'enzyme';

const [nameOfFunctionHere] = jest.fn();

describe('Show_Main', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Show_Main />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      // Put state here
    });
  });
})