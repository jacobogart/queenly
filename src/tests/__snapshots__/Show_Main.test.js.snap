// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show_Main should match snapshop 1`] = `
<main
  className="Bar-Main Main_Info"
>
  <article
    className="Content"
  >
    <h2>
      Sunday Social
    </h2>
    <p
      className="hosts"
    >
      Host: Chamilla Foxx
    </p>
    <p>
      When: 
    </p>
    <p
      className="showtime"
    >
      Every Sudnay on the weekly of the month.
       
    </p>
    <p
      className="time"
    >
      At: 
    </p>
  </article>
</main>
`;
